base/src/dispatcher/dispatcher.ts

```diff
@@ -372,16 +411,97 @@ export class Dispatcher implements ZLUX.Dispatcher {
     
 
 /* what will callback be called with */
-  registerAction(action: Action):void {
-      this.actionsByID.set(action.id,action);
-   }
+  registerAction(action: AbstractAction): void {
+    const { id } = action;
+    const existingAction: AbstractAction | undefined = this.actionsByID.get(id);
+
+    if (existingAction) {
+      this.log.warn(`duplicate actionId ${id}. Replacing`, existingAction, 'with', action);
+    }
+
+    this.actionsByID.set(action.id, action);
+
+    if (action instanceof ActionContainer) {
+      const { children } = action as ActionContainer;
 
+      if (children && children.length > 0) {
+        for (let child of children) {
+          if (child instanceof AbstractAction) {
+            this.registerAction(child);
+          }
+        }
+      }
+    }
+  }
 
-   getAction(recognizer:any):Action|undefined {
+   getAction(recognizer:any):AbstractAction|undefined {
```

Comment:
We need a jsdoc code comment to say why people shouldnt really use this anymore... and make it return an Action instead of an abstract one, because the purpose of this function was to then be able to do invokeAction() on it soon after.

Reply:
And so, if it cant get an Action, but gets another type of AbstractAction, it can return undefined with a console warning about why it was wrong